src/root/usr/bin/run-postgresql

```diff
@@ -20,8 +20,14 @@ generate_postgresql_config
 # Is this brand new data volume?
 PG_INITIALIZED=false
 
-if [ ! -f "$PGDATA/postgresql.conf" ]; then
-  initialize_database
+if [ ! -f "${PGDATA}/postgresql.conf" ]; then
+  # Minimize the race condition of unfinished DB initialization vs pod restart
+  # First initialize the DB into temporary directory
+  TEMP_PGDATA=$(mktemp -d $(printf '%0.sX' {1..32}))
+  initialize_database "${TEMP_PGDATA}"
+  # If DB initialization succeeded, copy content of temporary directory to
+  # final PGDATA, remove the temporary directory & set PG_INITIALIZED flag
+  cp -prT "${TEMP_PGDATA}" "${PGDATA}" && rm -rf "${TEMP_PGDATA}"
   PG_INITIALIZED=:
 else
   try_pgupgrade
```

Comment:
@pkubatrh (brainstorm) if "initializing" has a problems with timeouts, how `try_pgupgrade` is affected?